receiver/vmwarevcenterreceiver/scraper.go

```diff
@@ -0,0 +1,364 @@
+	if v.vsanEnabled {
+		hostVsanCSVs, err = v.client.CollectVSANHosts(ctx, &clusterRef, time.Now(), time.Now())
+		if err != nil {
+			errs.AddPartial(1, err)
+		}
+	}
+
+	for _, h := range hosts {
+		v.collectHost(ctx, colTime, h, hostVsanCSVs, errs)
+		v.mb.EmitForResource(
+			metadata.WithVcenterHostName(h.Name()),
+			metadata.WithVcenterClusterName(cluster.Name()),
+		)
+	}
+}
+
+func (v *vcenterMetricScraper) collectHost(
+	ctx context.Context,
+	now pdata.Timestamp,
+	host *object.HostSystem,
+	vsanCsvs *[]types.VsanPerfEntityMetricCSV,
+	errs *scrapererror.ScrapeErrors,
+) {
+	var hwSum mo.HostSystem
+	err := host.Properties(ctx, host.Reference(),
+		[]string{
+			"config",
+			"summary.hardware",
+			"summary.quickStats",
+		}, &hwSum)
+
+	if err != nil {
+		errs.AddPartial(1, err)
+		return
+	}
+	v.recordHostSystemMemoryUsage(now, hwSum)
+	v.recordHostPerformanceMetrics(ctx, hwSum, errs)
+	if vsanCsvs != nil {
+		entityRef := fmt.Sprintf("host-domclient:%v",
+			hwSum.Config.VsanHostConfig.ClusterInfo.NodeUuid,
+		)
+		v.addVSANMetrics(*vsanCsvs, entityRef, hostType, errs)
+	}
+}
+
+func (v *vcenterMetricScraper) collectResourcePools(
+	ctx context.Context,
+	ts pdata.Timestamp,
+	errs *scrapererror.ScrapeErrors,
+) {
+	rps, err := v.client.ResourcePools(ctx)
+	if err != nil {
+		errs.AddPartial(1, err)
+		return
+	}
+	for _, rp := range rps {
+		var moRP mo.ResourcePool
+		rp.Properties(ctx, rp.Reference(), []string{
+			"summary",
+			"summary.quickStats",
+			"name",
+		}, &moRP)
+		v.recordResourcePool(ts, moRP)
+		v.mb.EmitForResource(metadata.WithVcenterResourcePoolName(rp.Name()))
+	}
+}
+
+func (v *vcenterMetricScraper) collectVMs(
+	ctx context.Context,
+	colTime pdata.Timestamp,
+	cluster *object.ClusterComputeResource,
+	errs *scrapererror.ScrapeErrors,
+) {
+	vms, err := v.client.VMs(ctx)
+	if err != nil {
+		errs.AddPartial(1, err)
+		return
+	}
+
+	var vsanCsvs *[]types.VsanPerfEntityMetricCSV
+	if v.vsanEnabled {
+		clusterRef := cluster.Reference()
+		vsanCsvs, err = v.client.CollectVSANVirtualMachine(ctx, &clusterRef, time.Now().UTC(), time.Now().UTC())
+		if err != nil {
+			errs.AddPartial(1, err)
+		}
+	}
+
+	for _, vm := range vms {
+		var moVM mo.VirtualMachine
+		err := vm.Properties(ctx, vm.Reference(), []string{
+			"config",
+			"runtime",
+			"summary",
+		}, &moVM)
+
+		if err != nil {
+			errs.AddPartial(1, err)
+			continue
+		}
+
+		vmUUID := moVM.Config.InstanceUuid
+		entityRefID := fmt.Sprintf("virtual-machine:%s", vmUUID)
+		ps := string(moVM.Runtime.PowerState)
+
+		host, err := vm.HostSystem(ctx)
+		if err != nil {
+			errs.AddPartial(1, err)
+			return
+		}
+		hostname, err := host.ObjectName(ctx)
+		if err != nil {
+			errs.AddPartial(1, err)
+			return
+		}
+
+		v.collectVM(ctx, colTime, moVM, entityRefID, vsanCsvs, errs)
+		v.mb.EmitForResource(
+			metadata.WithVcenterVMName(vm.Name()),
+			metadata.WithVcenterVMID(vmUUID),
+			metadata.WithVcenterVMPowerState(ps),
```

Comment:
I don't believe this is identifying information for the resource, so shouldn't be a resource attribute.